### ISSUE TITLE
[Question/Feedback]: No VMs resources are listed as compliant.

### ISSUE THREAD

**Issue by jpgutton** [Azure/azure-monitor-baseline-alerts#138]
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Description

I am working on deploying AMBA, I tried deploying to different tenants, using aligned and unaligned deployment, either through pipeline or CLI, everything is working great except for all the VM policies which marks all as non-compliant. VMs where created an hour after AMBA deployment, remediation task gives nothing except the errors below.

<img width="430" alt="image" src="https://github.com/Azure/azure-monitor-baseline-alerts/assets/43048972/224f7194-a282-43ba-82e6-695b42e87e95">

<img width="1178" alt="image" src="https://github.com/Azure/azure-monitor-baseline-alerts/assets/43048972/fa8225a7-d93a-46f3-b80f-c25cda3f7509">

**Comment by Brunoga-MS** [comment]
hello @jpgutton ,
thanks for your feedback. I would like to understand more about this issue, since it looks like there was something wrong during the deployment. Could you please share the exact steps and order of execution?

Thanks,
Bruno.

**Comment by jpgutton** [comment]
I redid a test, clean everything, removed all existing policies:

![image](https://github.com/Azure/azure-monitor-baseline-alerts/assets/43048972/5fe99ab9-5db1-438b-9d8e-aa88227fac89)

,delete all MG, and most of the existing resources in the target subs, create a new MG hierarchy using the contoso one from the doc. 

<img width="733" alt="image" src="https://github.com/Azure/azure-monitor-baseline-alerts/assets/43048972/1a2141f6-9187-4f35-ba1d-4bb587f945cc">

I use the CLI to deploy the template, the only change I did in the param file was to set the location to Canada Central.

![image](https://github.com/Azure/azure-monitor-baseline-alerts/assets/43048972/266116b0-9386-412d-ad2b-15905468e0f2)


![image](https://github.com/Azure/azure-monitor-baseline-alerts/assets/43048972/4c284360-e6cf-4fd6-bb2e-e12ee114a77f)

I then deploy some resources like VNet etc and the policies went well. 

<img width="803" alt="image" src="https://github.com/Azure/azure-monitor-baseline-alerts/assets/43048972/da93cfe4-bc0f-4c29-9b27-8552ecefc199">

When I deployed a VM from the portal (next next next...) it didn't work. I waited then run the remediation script and the result is the same.

<img width="816" alt="image" src="https://github.com/Azure/azure-monitor-baseline-alerts/assets/43048972/aab4eb72-6d02-488c-b022-378214336070">

![image](https://github.com/Azure/azure-monitor-baseline-alerts/assets/43048972/b68a5512-2bdb-4ade-b758-df5c2188fedc)

![image](https://github.com/Azure/azure-monitor-baseline-alerts/assets/43048972/2a2f3bec-7ce0-4c11-9bbc-c5e0a7548d51)

![image](https://github.com/Azure/azure-monitor-baseline-alerts/assets/43048972/7762754b-d534-48e0-a563-8db33bec4ab3)


Note: I deployed it also in an unaligned hierarchy in my AIRS sub and after 2 remediations it works for the VM (existing and new).

**Comment by jpgutton** [comment]
I have PAYG subs, I'll clean everything and will redo the test.

**Comment by Brunoga-MS** [comment]
@jpgutton, could you please check if you have the VMInsights enabled? It was failing for me as well with the same error on a machine that was not onboarded into VMInsights; I enabled VMInsights and after the data appeared in the LAW I re-run the remediation and it went through. Was it maybe the case of your AIRS as well?

**Comment by Brunoga-MS** [comment]
Hello @jpgutton ,
I run my tests once again without any workspace or VMInsights onboarding and I was able to remediate VMs. I think it was just a sync issue going on for some time (like the policy compliance async results).

Let me know if tests on your side went good.

Thanks,
Bruno.

**Comment by jpgutton** [comment]
Hello, @Brunoga-MS 
I cleaned up everything and redeploy using PAYG subs, yesterdays I had mixed results, even after remediation, some VMs were seen as compliant some other not with the same error (AMA deployed by policy on one sub, VM Insight by hand on a second one).
This morning, I deployed new VMs in the same subs and all new ones are all green this PM.

Not sure why, maybe some latency in the policies. Ièll close it, will do more test next week in a new environment.
